pkg/pillar/hypervisor/xen.go

```diff
@@ -829,3 +832,45 @@ func execWithTimeout(command string, args ...string) ([]byte, bool, error) {
 	}
 	return out, true, err
 }
+
+//verifyDomainState: Verifies if a domain is up and running by its state. Domain can have the following states:
+//r - currently running
+//b - blocked, and not running or runnable
+//p - paused
+//s - a shutdown command has been sent, but the domain isn't dying yet
+//c - the domain has crashed
+//d - the domain is dying, but hasn't properly shut down or crashed
+func isDomainRunning(domainID int) bool {
+	cmd := "xl"
+	args := []string{
+		"list",
+		strconv.Itoa(domainID),
+	}
+	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		//domain is not present
+		log.Errorln("verifyDomainState: xl list failed ", err)
+		log.Errorln("verifyDomainState: xl list output ", string(stdoutStderr))
+		return false
+	}
+	//Removing all extra space between column result and split the result as array.
+	xlDomainResult := regexp.MustCompile(`\s+`).ReplaceAllString(strings.Split(string(stdoutStderr), "\n")[1], " ")
+	//Domain's status is 5th column in xl list <domain> result
+	domainState := strings.Split(xlDomainResult, " ")[4]
+	//Sometimes the domain can be in multiple states (logically possible) at the same time.
+	// In such cases we can consider the last state. For example when a domain is shutting down its state will be
+	//"--psc-", we can consider "c" to represent that domain is in middle of shutting down.
+	var lastState string
+	for _, state := range domainState {
+		if state != '-' {
+			lastState = string(state)
+		}
+	}
+	log.Infof("verifyDomainState: domain: %d domainState: %s.", domainID, domainState)
```

Comment:
This is called periodically hence the log.Info will store gigabytes per device in Kibana. Make it into a single log.Debug instead.

Reply:
Changed.